timeseries-forecast-daily/deaths-on-cases/ts-deaths-on-cases-daily.R

```diff
@@ -0,0 +1,164 @@
+# Forecast deaths from cases using timeseries
+# See <timeseries_death_forecast> for arguments
+# 
+# Additional arguments:
+# case_data : dataframe with case data
+# deaths_data : dataframe with deaths data
+# case_quantile : numeric, which quantile to return from case forecasting to be used in the death forecast
+# 
+# 
+# Example parameters:
+# deaths_data <- get_us_deaths(data = "daily")
+# case_data <- get_us_cases(data = "daily")
+# case_quantile = 0.5
+# sample_count = 1000
+# horizon_weeks = 7
+# right_truncate_weeks = 1
+# quantiles_out <- c(0.01, 0.025, seq(0.05, 0.95, by = 0.05), 0.975, 0.99)
+
+library(magrittr); library(dplyr); library(tidyr); library(EpiSoon); library(forecastHybrid)
+
+ts_deaths_on_cases_forecast <- function(case_data, deaths_data, case_quantile,
+                                           sample_count, horizon_days, right_truncate_days){
+    
+
+# Set up cases ----------------------------------------------------------
+
+  # Set up case data and truncate
+  
+  right_truncate_date <- max(case_data$date) - right_truncate_days
+  
+  case_data <- case_data %>%
+    filter(epiweek <= right_truncate_date) 
+  
+
+# Set up dates -----------------------------------------------------------
+
+  # Forecasting = next week + horizon_weeks
+  forecast_days <- seq(from = max(case_data$date)+1, by = 1, length.out = horizon_days + right_truncate_days)
+  case_forecast_days <- seq(from = max(case_data$date)+1, by = 1, length.out = horizon_days + right_truncate_days)
+  
+  # Historical = last 6 weeks of data
+  historical_days <- case_data %>%
+    ungroup() %>%
+    filter(!date %in% forecast_days) %>%
+    select(date) %>%
+    filter(date > max(date) - (6*7) &
+             date <= right_truncate_date) %>%
+    unique() %>%
+    pull(date)
+
+  
+
+# Forecast cases ---------------------------------------------------------
+   
+  case_forecast <- case_data %>%
+      group_by(state) %>%
+      group_modify(~ EpiSoon::forecastHybrid_model(y = filter(.x, date %in% historical_days) %>%
+                                                     pull("cases"),
+                                                   samples = sample_count, 
+                                                   horizon = horizon_days + right_truncate_days,
+                                                   model_params = list(models = "aez", weights = "equal",
```

Comment:
Suggest review model list and weights options (suggest moving to a weighted ensemble potentially or at least testing).